### ISSUE TITLE
0x24 ($) is not a valid escaped character

### ISSUE THREAD

**Issue by Skeeve** [ohler55/ojg#164]
Hi!

When I use `$..[?(@ =~ /^\$/)]` I get "0x24 ($) is not a valid escaped character" while when I use `$..[?(@ =~ /^[$]/)]` it works as expected.

Not-working example

```go
package main

import (
	"fmt"

	"github.com/ohler55/ojg/jp"
	"github.com/ohler55/ojg/oj"
)

func main() {
	data, err := oj.ParseString(`
    {
        "type": "AdaptiveCard",
        "body": [
            {
                "type": "Container",
                "isVisible": "$._has_content.success",
                "items": [
                    {
                        "type": "TextBlock",
                        "text": "$.result[0].text"
                    },
                    {
                        "type": "FactSet",
                        "facts": [
                            "$.success.*"
                        ]
                    }
                ]
            },
            {
                "type": "Container",
                "style": "attention",
                "isVisible": "$._has_content.fail",
                "items": [
                    {
                        "type": "TextBlock",
                        "text": "Failures"
                    },
                    {
                        "type": "FactSet",
                        "facts": [
                            "$.fail.*"
                        ]
                    }
                ]
            }
        ]
    }
	`)
	if err != nil {
		fmt.Println(err.Error())
		return
	}
	
	x, err := jp.ParseString(`$..[?(@ =~ /^\$/)]`)
	if err != nil {
		fmt.Println(err.Error())
		return
	}
	fmt.Println(x.String())
	result := x.Get(data)
	fmt.Println(oj.JSON(result))
}
```

Working example on [go playground](https://play.golang.com/p/5wUP0JC10zx)

**Comment by Skeeve** [comment]
**Update:** Same applies for `\.` vs. `[.]`.

**Comment by ohler55** [comment]
I have a pretty good idea what is wrong but will need a few days to fix it.

**Comment by Skeeve** [comment]
Perhaps both are missing as cases in this switch? https://github.com/ohler55/ojg/blob/a1754591ad7c7c8af2093e2c1f6273811df24564/jp/parse.go#L458

**Comment by ohler55** [comment]
A new read function will be needed since regexp have different rules regarding allowable escape sequences.

**Comment by ohler55** [comment]
Please try the better-regex-parsing branch.

**Comment by Skeeve** [comment]
I'd love to, Peter. But as I'm quite new to GO development, I'm not sure, how to do that.

Do I have to clone the repo? Or can I point my import section to that branch?

**Comment by ohler55** [comment]
I'll release today.

**Comment by ohler55** [comment]
Released v1.21.4
